### PR TITLE
workers updates to use socrata-docker cookbook

### PR DESCRIPTION
**Pull request by JoeNunnelley**
- convert to using socrata-docker to deploy docker and docker configuration to workers
- add some tests
- reformat includes 
- include .kitchen.yml with docker related test configuration
- convert to using dokken as that is what socrata-docker uses and it appears I have to mirror that. (Jon, let me know if that is get-around-able. It seemed to not be)

### CODE REVIEW

**Comment by JoeNunnelley** [comment]
retest this please

File: Berksfile [hunk 1]
```diff
@@ -5,9 +5,4 @@ source artifactory: 'https://repo.socrata.com/artifactory/api/chef/chef'
 metadata
 
 cookbook 'sbt', git: 'git@github.com:socrata-cookbooks/sbt'
-cookbook 'golang', '~> 1.7.0'
-cookbook 'java', '~> 1.50.0'
-cookbook 'nodejs', '~> 5.0.0'
-cookbook 'apt', '~> 5.0.0'
-cookbook 'docker', '~> 2.13.0'
-cookbook 'postgresql', '~> 5.2.0'
+cookbook 'socrata-docker', git: 'git@github.com:socrata-cookbooks/socrata-docker'
```
Comment (RoboticCheese) [inline Berksfile]: Why point at Github when it lives in artifactory like all our other cookbooks?

File: Gemfile [hunk 2]
```diff
@@ -1,8 +1,36 @@
 source 'https://repo.socrata.com/artifactory/api/gems/rubygems-virtual/'
 
 gem 'berkshelf'
-gem 'test-kitchen'
-gem 'kitchen-docker'
-gem 'foodcritic'
 gem 'chef', '~> 12.20.0'
 gem 'molinillo', '< 0.6.0'
```
Comment (RoboticCheese) [inline Gemfile]: FYI, the molnillo pin is no longer needed. That bug was fixed.

File: .kitchen.yml [hunk 3]
```diff
@@ -1,23 +1,101 @@
...
+    attributes:
+      s3_build_artifacts:
+        aws_region: <%= region %>
+        aws_credential_bag: aws
+        aws_credential_item: <%= env %>
+        bucket_name: socrata-build-artifacts-<%= region %>
+      docker:
+        bip: 172.16.42.1/16
+        dns: 172.16.42.1
+        ecr:
```
Comment (RoboticCheese) [inline .kitchen.yml]: All these ecr attributes were just stub data for testing socrata-docker. They should be safe to delete.

File: attributes/default.rb [hunk 4]
```diff
@@ -3,6 +3,9 @@
 # Set the ubuntu user to be a member of the rbenv group
 default['worker_user'] = 'ubuntu'
 
+# Awd configuration
+default['aws']['region'] = 'us-west-2'
```
Comment (RoboticCheese) [inline attributes/default.rb]: Where does this attribute end up getting used? I didn't see references to it anywhere else in this PR.

File: Berksfile [hunk 5]
```diff
@@ -4,10 +4,5 @@ source artifactory: 'https://repo.socrata.com/artifactory/api/chef/chef'
...
 metadata
 
-cookbook 'sbt', git: 'git@github.com:socrata-cookbooks/sbt'
-cookbook 'golang', '~> 1.7.0'
-cookbook 'java', '~> 1.50.0'
-cookbook 'nodejs', '~> 5.0.0'
-cookbook 'apt', '~> 5.0.0'
-cookbook 'docker', '~> 2.13.0'
-cookbook 'postgresql', '~> 5.2.0'
+cookbook 'sbt'
```
Comment (RoboticCheese) [inline Berksfile]: These deps are already listed in `metadata.rb` so are redundant here.

File: recipes/_docker.rb [hunk 6]
```diff
@@ -1,8 +1,9 @@
-include_recipe 'apt'
+user node['worker_user']
 
-docker_service 'default' do
-  storage_driver 'aufs'
-end
+include_recipe 'socrata-docker'
+#include_recipe 'socrata-docker::install'
```
Comment (RoboticCheese) [inline recipes/_docker.rb]: What's going on with the commented-out `include_recipe`s here?

**Pull request merged**